### PR TITLE
Don't resolve _ensureRegistered if we cannot ensure a user is registered.

### LINKED ISSUE #71: _ensureRegistered() will always be resolved, even when failed.

### PR DESCRIPTION
**Pull request by Half-Shot**
I think that's fairly sane?

Fixes #71

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "env": {
     "node": true,
-    "jasmine": true
+    "jasmine": true,
+    "es6": true,
```
Comment (turt2live) [inline .eslintrc]: I think this is being dealt with elsewhere?

File: lib/components/intent.js [hunk 2]
```diff
@@ -649,17 +649,17 @@ Intent.prototype._ensureRegistered = function() {
...
+    const localpart = new MatrixUser(userId).localpart;
+    return self.botClient.register(localpart).then((res) => {
+        this.opts.registered = true;
         return res;
-    }, function(err) {
+    }).catch((err) => {
         if (err.errcode === "M_USER_IN_USE") {
-            self.opts.registered = true;
+            this.opts.registered = true;
             return err;
```
Comment (turt2live) [inline lib/components/intent.js]: If we're rejecting the promise, we should probably suppress this error or let it fall through to the throw below, whichever is best.

Comment (Half-Shot) [reply lib/components/intent.js]: Supression works for me.

File: .eslintrc (hunk 1 above)
Comment (Half-Shot) [reply .eslintrc]: This was to stop it from bitching about consts and arrow funcs, sorry. We've already merged this change in, so it should be fine.

File: lib/components/intent.js [hunk 3]
```diff
@@ -649,17 +649,17 @@ Intent.prototype._ensureRegistered = function() {
...
+    return self.botClient.register(localpart).then((res) => {
+        this.opts.registered = true;
         return res;
-    }, function(err) {
+    }).catch((err) => {
         if (err.errcode === "M_USER_IN_USE") {
-            self.opts.registered = true;
-            return err;
+            this.opts.registered = true;
+            return {};
```
Comment (turt2live) [inline lib/components/intent.js]: I think you can return null and get the same effect. Objects feel a bit weird to me, and would appreciate a comment if kept.

**Pull request merged**